### PR TITLE
Add post on existentials.

### PR DESCRIPTION
**Pull request by patrickt**

### CODE REVIEW

File: posts/existential-haskell.org [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+The built-in ~~~ syntax, included with the ~GADTs~ extension, provides us a method to establish that type variables must be equal: the present of an ~a ~ Int~ constraint ensures that ~a~ /must/ unify with (read: be equal to) the ~Int~ type. By partially applying this constraint, we can speak of useful types with remarkable brevity, such as the following type representing “some ~Functor~ containing ~Int~ values”:
+
+#+BEGIN_SRC haskell
+someFunctorOfInts :: Some1 Functor ((~) Int)
+someFunctorOfInts = Some1 [1, 2, 3]
+#+END_SRC
+
+** Conclusion
+
+Draw your own conclusions. We’re done here. Thanks very much for reading.
```
Comment (aymannadeem) [inline posts/existential-haskell.org:319]: The tone of this conclusion feels abruptly abrasive.

File: posts/existential-haskell.org [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+  Some :: c a => a -> Some c
+#+END_SRC
+
+With the ~TypeApplications~ and ~ConstraintKinds~ extensions, we can specify that the type variable passed to ~Some~ is not of kind ~Type~, or ~Type -> Type~; instead, it takes a ~Type~ and returns a ~Constraint~. This means that we can pass in ~Show~, ~Eq~, ~Ord~, or any other unary typeclass, using a type application:
+
+#+BEGIN_SRC haskell
+let wrappedInt = Some @Show (5 :: Int)
+#+END_SRC
+
+This seems like a broadly applicable data type, but I haven’t seen anyone use anything like it yet (though the inimitable Rob Rix tells me that he’s defined it many times, at which I bear zero surprise, because Rob is a maestro).
```
Comment (aymannadeem) [inline posts/existential-haskell.org:293]: the inimitable Rob Rix strikes again!

File: posts/existential-haskell.org [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+
+What follows is an overview of existential types in practice, and the quirks associated therewith. I write this not because I expect to break any new ground—all the techniques I use here are long-documented in the literature, and Haskell veterans will probably find little new in this post[fn:0]—but because the existing resources are scattered, perhaps oddly so given how central dynamic dispatch is to most programming languages that aren't Haskell, and because exploring the edge cases in the design illustrates the compromises inherent in language and library design.
+
```
Comment (aymannadeem) [inline posts/existential-haskell.org:29]: I love this and would kinda prefer it upfront? I like how you hook readers in with the comparison to Java being loosey goosey with interfaces vs. concrete types, but as a reader I prefer to know upfront what I'm getting into. Another idea is to break up the above section with more sub-headings.

This is also biased by the accessibility required for my brain, where I need to constantly remind myself what I am doing, what I am reading, to make sure I am tracking, so if that suggestion totally kills your vibe and art, I respect that.

Comment (patrickt) [reply posts/existential-haskell.org:29]: Hmm, let me think about this. I think I like the hook, but maybe I can split it into two parts, or a summary header.

**Pull request merged**